src/cli/shim.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{Parser};
+use std::{env, process::{Command}};
+
+use crate::cli::common::get_default_araki_bin_dir;
+
+#[derive(Parser, Debug)]
+#[command(arg_required_else_help = true)]
+pub struct Args {
+    /// Invocation of another env management tool, e.g. `pip install foo`
+    #[arg(num_args = 1..)]
+    args: Vec<String>
+}
+
+
+
+/// Given a PATH environment variable, this function strips out the araki bin directory.
+///
+/// * `path`: Colon-separated PATH environment variable to be stripped
+fn strip_araki_shim_path(path: &str) -> Result<String, String> {
+    let araki_bin_dir = get_default_araki_bin_dir()?;
+    Ok(
+        path
+            .split(":")
+            .skip_while(|item| **item == araki_bin_dir)
+            .collect::<Vec<&str>>()
+            .join(":")
+    )
+}
+
+pub fn execute(args: Args) {
+    let value = env::var("ARAKI_OVERRIDE_SHIM").unwrap_or("false".to_string());
+    if value.trim() == "1" {
+        // Run the requested command using the modified PATH
+        let current_path = env::var_os("PATH");
+
+        // Extract the tool to be run `pip`, etc... from the argument list passed to araki.
+        // Call the tool and pass in any trailing arguments using the stripped PATH env variable.
+        if let [tool, arguments @ ..] = args.args.as_slice() {
+            let mut command = Command::new(tool);
+            if let Some(path) = current_path {
+                match strip_araki_shim_path(&path.to_string_lossy()) {
+                    Ok(new_env) => command.env("PATH", new_env),
+                    Err(err) => {
+                        eprintln!("Unable to strip the araki shim path from PATH:\n{err}");
+                        return;
+                    }
+                };
+            }
+            let _ = command
```

Comment:
I expect running the `araki shim` command allows a user to invoke the shimed command.

Probably need to export the `ARAKI_OVERRIDE_SHIM` here or remove the araki bin dir from path while executing the command. Otherwise this does not allow passing executing the command.

For example, the current behaviour is
```
 araki shim -- pixi -h
Unable to run pixi -h; use araki for environment management. Set ARAKI_OVERRIDE_SHIM=1 to run the command anyway.

```

To make it work as I expect I need to run:
```
$ ARAKI_OVERRIDE_SHIM=1 araki shim -- pixi -h
. . .
```

Reply:
After a discussion, `araki shim` is intended only to be called by a shim script, and it lets `araki` know the user is trying to call an unsupported env management tool. The documentation for this command has been updated.